tests/FunctionMapTest.php

```diff
@@ -109,6 +109,30 @@ public function shouldResolveToEmptyArrayWhenInputPromiseDoesNotResolveToArray()
     }
 
     /** @test */
+    public function shouldPreserveTheOrderOfArrayWhenResolvingAsyncPromises()
+    {
+        $queue = new \SplQueue();
+        $asyncPromise = new Promise(function ($resolve) use ($queue) {
+            $queue->enqueue(function () use ($resolve, $queue) {
+                $resolve(2);
+            });
+        });
+
+        $result = null;
+
+        map(
+            [resolve(1), $asyncPromise, resolve(3)],
+            $this->mapper()
+        )->then(function ($resolvedValues) use (&$result) {
+            $result = $resolvedValues;
+        });
+
+        call_user_func($queue->dequeue());
+
+        $this->assertSame([2 ,4 ,6], $result);
+    }
+
+    /** @test */
```

Comment:
I'd like to see the style of the test matching the rest of the tests (see my comment on the `all` test).